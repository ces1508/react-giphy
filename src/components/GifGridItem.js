import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({ url, title }) => {
  return (
    <div className="card">
      <div className="card-image">
        <img
          src={url}
          alt={title} />
      </div>
      <p>{title}</p>
    </div>
  );
}

GifGridItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}


export default GifGridItem