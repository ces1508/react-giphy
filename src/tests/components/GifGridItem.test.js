import React from 'react'
import { shallow } from 'enzyme'
import GifGridItem from '../../components/GifGridItem'

describe('<GifGridItem />', () => {

  const titleProps = 'title of image'
  const urlImageProps = 'https://image.com/example.jpg'

  const wrapper = shallow(
    <GifGridItem
      title={titleProps}
      url={urlImageProps} />
  )


  it ('renders <GifGridItem />', () =>  {
    expect(wrapper).toMatchSnapshot()
  })

  it('<GifGridItem /> should be have a p with title', () => {
    const p = wrapper.find('p')
    expect(p.text().trim()).toBe(titleProps)
  })

  it ('<GifGridItem /> should be have an image with the url and alt send on props', () => {
    const image = wrapper.find('img')
    expect(image.prop('src')).toBe(urlImageProps)
    expect(image.prop('alt').trim()).toBe(titleProps)
  })

  it ('<GifGridItem /> should be have class card', () => {
    const divWrapper = wrapper.find('div').at(0)
    expect(divWrapper.hasClass('card')).toBe(true)
  })

  it ('<GifGridItem /> should be have class card-image', () => {
    const divWrapperImage = wrapper.find('div').at(1)
    expect(divWrapperImage.hasClass('card-image')).toBe(true)
  })
})
