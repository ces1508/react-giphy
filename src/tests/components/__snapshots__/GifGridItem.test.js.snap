// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> renders <GifGridItem /> 1`] = `
<div
  className="card"
>
  <div
    className="card-image"
  >
    <img
      alt="title of image"
      src="https://image.com/example.jpg"
    />
  </div>
  <p>
    title of image
  </p>
</div>
`;
